WordPress/gutenberg
# [Multi-selection]: Copy/cut partial selected blocks

## What?
Fixes: https://github.com/WordPress/gutenberg/issues/39455
This PR implements `copy/cut` functionality for partial selections.


## Why?
Currently if we copy/cut when we have a partial selection, the whole blocks are affected(copied/cut). We need to only copy/cut the partial selection.


## Testing Instructions
1. Make partial selections and copy or cut either from the the keyboard shortcuts (cmd + c/x) or the `copy` button from block actions.
2. Paste
3. Observe that only the partial selection is copied/cut/pasted

## Notes
1. ~~I'll have to write e2e tests~~ ✅ 
2. Some tidying up needs to be done in general, but some of it might be in followups to reuse functionality or move some things to be `actions`, etc...
3. If we have a partial selection that is not mergeable we fallback to just expanding the selection to the whole blocks.

## Screenshots or screencast 

https://user-images.githubusercontent.com/16275880/161990600-3fe20d04-7c67-4c66-81a3-7444df1bc1aa.mov
